static/js/components/dashboard/CourseSubRow.js

```diff
@@ -124,11 +125,21 @@ export default class CourseSubRow extends React.Component {
         />;
       }
     }
+    let status = '';
+    if ( courseRun.status === STATUS_NOT_PASSED ) {
+      status = 'Failed';
+    } else if ( courseRun.status === STATUS_PASSED ) {
+      status = 'Passed';
+    }
     return <div className="course-action">
-      { courseRun.status === STATUS_NOT_PASSED ? 'Failed' : '' }
+      { status }
```

Comment:
this method is a little hairy now (it was a little hairy before, but is somewhat more hairy with these changes). i think it's appropriate to split this up at this point. something roughly like this:

```javascript
renderAction(): React$Element<any> {
  if (courseRun.status === STATUS_OFFERED) {
    return this.renderOfferedCourseAction();
  else if (this.isCompletedCourseRun(courseRun)) {
    return this.renderGradeStatus(); // or just copy lines 128 to 136 here...
  }
  else {
    return null;
  }
}
```